db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -1183,14 +1183,20 @@ public boolean existsSchemaOwnedBy(String authid,TransactionController tc) throw
     public void addColumnStatistics(ExecRow row,
                               TransactionController tc) throws StandardException{
         TabInfoImpl ti=getNonCoreTI(SYSCOLUMNSTATS_CATALOG_NUM);
-        ti.insertRow(row,tc);
+        int insertRetCode = ti.insertRow(row,tc);
+        if (insertRetCode != TabInfoImpl.ROWNOTDUPLICATE)
+            throw StandardException.newException(SQLState.LANG_DUPLICATE_KEY_CONSTRAINT,
+                    "SYSCOLUMNSTATISTICS_INDEX1", SYSCOLUMNSTATISTICSRowFactory.TABLENAME_STRING);
```

Comment:
you could refactor this in a method called `insertOrThrow` and reuse it everywhere :)

Reply:
The thrown exception, number of parameter, and error code are not the same. Need to write at least three versions of insertOrThrow. I may do this later if that can make code much cleaner.